### PR TITLE
improve object logging for verify calls

### PR DESCRIPTION
**Pull request by mikeporterdev**
Prints better logging for the "Actual Calls" section for objects.
Previously we just printed `methodName([object Object])` whereas this will now JSON stringify.
Adds a dependency for a safe JSON stringifier in order to handle circular dependencies. Let me know if you'd prefer to avoid dependencies and I can try and pull similar logic in house.

I'd also like to see if we can add some nice colour comparison for the console, similar to Jest, but will look at that as another PR because that's useful outside of this logger

### CODE REVIEW

**Comment by mikeporterdev** [comment]
@cspotcode @LironHazan Could I get a review on this one? Happy to volunteer to help out with maintaining where possible

**Comment by LironHazan** [comment]
@mikeporterdev hey I just noticed :) sure I'll review!

File: package.json [hunk 1]
```diff
@@ -64,7 +65,8 @@
     "typescript": "^4.7.4"
   },
   "dependencies": {
-    "lodash": "^4.17.5"
+    "lodash": "^4.17.5",
+    "safe-json-stringify": "^1.2.0"
```
Comment (LironHazan) [inline package.json:69]: I need to plugin a snyk workflow.. this tiny dep won't introduce any vulns (it has no dependencies)

File: src/utils/MethodCallToStringConverter.ts [hunk 2]
```diff
@@ -9,6 +10,16 @@ export class MethodCallToStringConverter {
     }
 
     public convertActualCalls(calls: MethodAction[]): string[] {
-        return calls.map(call => `${call.methodName}(${call.args.map(arg => arg.toString()).join(", ")})`);
+        return calls.map(call => {
+            const methodName = call.methodName;
+            const args = call.args.map(arg => {
+                if (typeof arg !== 'object' || arg.hasOwnProperty('toString')) {
```
Comment (LironHazan) [inline src/utils/MethodCallToStringConverter.ts:16]: can we move this --> ```typeof arg !== 'object' || arg.hasOwnProperty('toString')``` to be a predicate? something like:
`.map(arg =>  shouldUseSafeStringify() ? safeJsonStringify(arg) : arg.toString()`

File: test/matcher/type/DeepEqualMatcher.spec.ts [hunk 3]
```diff
@@ -86,21 +86,80 @@ describe("DeepEqualMatcher", () => {
             // when
             const result = testObj.match(secondValue);
 
+            // then
+            expect(result).toBeFalsy();
```
Comment (LironHazan) [inline test/matcher/type/DeepEqualMatcher.spec.ts:90]: I can't see the context of this but I assume it isn't related to you additions? (it's in a scope of an older test)

File: test/utils/Foo.ts [hunk 4]
```diff
@@ -46,4 +46,8 @@ export class Foo {
     public async sampleMethodReturningVoidPromiseWithoutParams(): Promise<void> {
         return Promise.resolve();
     }
+
+    public async sampleMethodWithObjectArguments(obj: object): Promise<void> {
+        return;
```
Comment (LironHazan) [inline test/utils/Foo.ts:51]: we can return `void 0` if we want to be extra extra explicit :)

**Comment by LironHazan** [comment]
@mikeporterdev do you have a clue why the PR didn't trigger any checks? maybe due to inactivity the checks were disabled? checking

**Comment by mikeporterdev** [comment]
@LironHazan It tells me that contributors need to be approved by a maintainer for automatic checks 
image

I think it's a github step to tackle rampant cryptominers in Github Actions

Will take a look at your feedback this afternoon :) Thanks!

File: test/utils/Foo.ts (hunk 4 above)
Comment (mikeporterdev) [reply test/utils/Foo.ts]: Good call on being explicit. To stay consistent with the rest of the file I've added `return Promise.resolve()`

File: src/utils/MethodCallToStringConverter.ts (hunk 2 above)
Comment (mikeporterdev) [reply src/utils/MethodCallToStringConverter.ts]: Done :)

**Comment by LironHazan** [comment]
> @LironHazan It tells me that contributors need to be approved by a maintainer for automatic checks image
> 
> I think it's a github step to tackle rampant cryptominers in Github Actions
> 
> Will take a look at your feedback this afternoon :) Thanks!

Right! Done

**Comment by LironHazan** [comment]
@mikeporterdev Do you have another improvement to add?
If you'll bumpup the version on the next PR - both changes will be published with the newer version,
@cspotcode 
We hadn't discussed "newer" additions or what is the approach for publishing minor versions in aspect of content,
As I can't invest too much time in development of this project (just maintenance, bug fixes) I don't mind bumping up the minor version on every PR similar to this one (minor improvements)

**Comment by mikeporterdev** [comment]
> @mikeporterdev Do you have another improvement to add?
> If you'll bumpup the version on the next PR - both changes will be published with the newer version,
> @cspotcode 
> We hadn't discussed "newer" additions or what is the approach for publishing minor versions in aspect of content,
> As I can't invest too much time in development of this project (just maintenance, bug fixes) I don't mind bumping up the minor version on every PR similar to this one (minor improvements)

@LironHazan I think these currently open PRs are all I've got in the short term. Ideally longer term I'd like to start helping out more with maintenance, and by copying over more of the remaining work from the old repo into here

Once all my PRs are merged I don't mind opening a separate PR for bumping the minor version if that's how deployments are raised

**Comment by LironHazan** [comment]
@mikeporterdev Sounds good, what types of permissions should I give you so your PRs will auto trigger the actions? just write? I can add you as a maintainer gladly just want @cspotcode approval on that

**Comment by mikeporterdev** [comment]
Would be happy to be marked as a maintainer if that means I can help out and if approved :) Not sure exactly what permissions I would need, depending on the pipeline. 

If the pipeline deploys automatically on minor version changes then I shouldn't actually need any permission changes as long as one of the current maintainers approves and merges the version bump PR. But if marked as a maintainer I'd be happy to help with other issues and PRs as they come in

**Comment by mikeporterdev** [comment]
@LironHazan Sorry I forgot to @ you in my previous response :)

**Pull request merged**